soroban-env-host/src/native_contract/token/balance.rs

```diff
@@ -5,20 +5,33 @@ use crate::native_contract::token::public_types::{Identifier, Metadata};
 use crate::native_contract::token::storage_types::DataKey;
 use crate::{err, HostError};
 use soroban_env_common::xdr::{
-    AccountEntryExt, AccountEntryExtensionV1Ext, AccountId, LedgerEntryData, TrustLineAsset,
-    TrustLineEntryExt, TrustLineFlags,
+    AccountEntry, AccountEntryExt, AccountEntryExtensionV1Ext, AccountFlags, AccountId,
```

Comment:
I think this file should be refactored to separate trustline vs non-trustline logic. We can do that in a separate PR.